### DISCUSSION TITLE
Merging Inventory Sources

### DISCUSSION THREAD

**Discussion by agreusel** [nornir-automation/nornir#809] [Q&A]
Is there a 'baked in' mechanism for merging inventory data from multiple sources?

The use case is to use NetBox as the primary inventory. However, there are some things that can't be modeled in NetBox without writing a custom plugin, and we would prefer to store those in the more traditional SimpleInventory hosts.yml file.

What we'd like to do is load the inventory data for the same devices, from both sources (giving priority for duplicate data to NetBox), resulting in a single inventory to use for tasks.

I have a suspicion that this is not currently possible and we'll need to write a custom inventory plugin to do it. But I figured I'd ask here just in case.

Thanks!

**Comment by dbarrosop** [comment]
You can load any data you want into a host by using its `data` attribute. Most notably there is a `load_yaml` plugin you can already leverage. So load your inventory as usual and then create a task that loads any extra data you want.
